### DISCUSSION TITLE
How do I resize a `gt` table

### DISCUSSION THREAD

**Discussion by SarahJMills** [quarto-dev/quarto-cli#9612] [Q&A] [answered]
### Description

Hi

I've created a `gt` table that I want to render to pdf but was struggling to get all the styling rendered.  I saw a Rich Iannone post somewhere 4th July 2022 explaining that LaTeX tables in gt don't support the full set of styling options available in HTML tables, but that a reasonable work around was to save an HTML table as a PNG graphic with the `greave()` function.

That does seem to work, but I don't understand how this code, which I thought would just save the png to file, also causes it to be rendered in the pdf.

`gtsave(tbl_msiassay_gt, file.path(here("figures"), "fig_msiassay.png"))`

I therefore am not sure how/where to add code to play with the size of the PNG version of the table.

Grateful for any guidance.

**Comment by SarahJMills** [comment]
Sorry...typo above..., should read `gtsave()` at the end of the first paragraph.

**Reply by cscheid** [reply]
I don't know how `gt` and `gtsave` work (and maybe https://github.com/rstudio/gt is the best place to ask for help there)

With that said, you might be able to understand a bit better what's going on by adding `keep-md: true` to your document front matter and inspecting the `.pdf.md` file that will be left in your document's folder after `quarto render`.

**Reply by SarahJMills** [reply]
Many thanks, I'll try that and/or post to the link you provided.

**Comment by mcanouil** [comment, marked as answer]
It's a bit unclear what your Quarto document looks like without having the code.

FYI, you don't need `file.path` if you are using `here` (that's the point of using `here`).

```r
gtsave(tbl_msiassay_gt, here("figures", "fig_msiassay.png"))
```

The above only writes the image. For question regarding this function, as suggested, go to `gt` documentation/repository.

Now the image can be included in your Quarto document, using basic markdown syntax for figures, where you can set the dimension.


---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by SarahJMills** [reply]
Very many thanks for all the advice and apologies for offering insufficient information in the original question (I'm a clinician and coding for reproducible research is a steep learning curve for me).

1. Noted re not needing file.path & code amended
2. 'The above only writes the image'...that's what I thought too, but:

- including this line of code creates the file `"fig_msiassay.png"` in the `"figures"` folder AND causes the image to be rendered in the pdf. 
- commenting out that line of code does not create the file NOR does the image appear in the rendered pdf

3. Thank-you for the advice re creating a self-contained reproducible document and a Git repository, I hope to learn how to do that in the future, but this project is a dissertation using (anonymised) live patient data.  Ethical review required me to keep all files and information within the hospital trust IT systems.  I've saved this exchange and the links you provided to refer to later.

Much appreciate your time and trouble.

**Reply by mcanouil** [reply]
If the image is displayed, it's probably because `gtsave` returns the file path (https://gt.rstudio.com/reference/gtsave.html).

**Comment by baptiste** [comment]
Just as a side-note, I believe recent versions have introduced a grid graphics backend, so you could try that to embed a vector (pdf) image rather than a raster: https://gt.rstudio.com/reference/as_gtable.html?q=grid#ref-usage
Don't know the feature set coverage though

**Reply by SarahJMills** [reply]
Thank-you, will take a look at that link and experiment.
